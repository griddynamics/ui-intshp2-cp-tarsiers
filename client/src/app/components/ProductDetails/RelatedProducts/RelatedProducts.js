import React, { Component } from 'react';

import { productType } from '../../../types';
import Carousel from '../../../shared/Carousel/index';
import ProductItemContainer from '../../../shared/ProductItem/ProductItemContainer';
import HttpService from '../../../../utils/http.service';
import {
  isAddedToCart,
  isAddedToWishList
} from '../../../../utils/inCartInWishlist.service';
import './RelatedProducts.scss';
import appConfig from '../../../../config/appConfig';

const title = 'Related Products';
const { products } = appConfig.apiResources;

class RelatedProducts extends Component {
  static propTypes = {
    item: productType
  };

  static defaultProps = { item: null };

  constructor(props) {
    super(props);
    this.state = { relatedProducts: [] };
  }

  componentDidMount = () => {
    this.getRelatedProducts();
  };

  getRelatedProducts = () => {
    const { item } = this.props;

    const params = {
      category: item.category,
      tag: item.tag
    };

    HttpService.get(products, { params }).then(res => {
      this.setState({ relatedProducts: res.data });
    });
  };

  render() {
    const titleArr = title.split(' ');
    const { wishlist, cart } = this.props;
    const { relatedProducts } = this.state;
    const extended = true;

    const list =
      relatedProducts &&
      relatedProducts.map(el => (
        <ProductItemContainer
          isAddedToCart={isAddedToCart(el._id, cart)}
          isAddedToWishList={isAddedToWishList(el._id, wishlist)}
          extended={extended}
          key={el._id}
          data={el}
        />
      ));

    return (
      <section className="container related-products">
        <div className="section_heading">
          <h2>
            <span className="highlighted">{titleArr[0]}</span>
            &nbsp;
            {titleArr[1]}
          </h2>
          <p>
            Lorem Ipsum is simply dummy text of the printing and typesetting
            industry
          </p>
        </div>
        <Carousel data={relatedProducts} extended>
          {list}
        </Carousel>
      </section>
    );
  }
}

export default RelatedProducts;
