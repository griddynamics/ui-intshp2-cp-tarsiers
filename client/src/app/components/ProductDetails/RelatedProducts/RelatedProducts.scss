@import '../../../../styles/variables';
@import '../../../../styles/mixin';

.related-products .section_heading {
  padding: 115px 0 70px 0;
  margin: 0 auto;
  text-align: center;

  h2 {
    font-size: 28px;
  }
}
