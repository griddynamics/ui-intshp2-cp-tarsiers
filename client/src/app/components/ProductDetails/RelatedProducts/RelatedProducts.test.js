import React from 'react';
import { shallow } from 'enzyme';
import RelatedProducts from './RelatedProducts';

const products = require('../../../../mocks/products.json');

describe('<RelatedProducts />', () => {
  it('should match its snapshot', () => {
    const item = products[0];
    const wrapper = shallow(<RelatedProducts item={item} />);

    expect(wrapper).toMatchSnapshot();
  });
});
