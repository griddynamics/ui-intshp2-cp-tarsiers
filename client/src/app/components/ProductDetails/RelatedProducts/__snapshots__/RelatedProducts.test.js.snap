// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RelatedProducts /> should match its snapshot 1`] = `
<section
  className="container related-products"
>
  <div
    className="section_heading"
  >
    <h2>
      <span
        className="highlighted"
      >
        Related
      </span>
      Â 
      Products
    </h2>
    <p>
      Lorem Ipsum is simply dummy text of the printing and typesetting industry
    </p>
  </div>
  <Carousel
    data={Array []}
    extended={true}
  />
</section>
`;
